nix/python.nix

```diff
@@ -0,0 +1,110 @@
+        with lib;
+        let
+          raw = if pathExists pythonVersionFile then readFile pythonVersionFile else "";
+          m = match "([0-9]+)\\.([0-9]+).*" (removeSuffix "\n" raw); # ["3" "13"]
+        in
+        if m != null && m != [ ] then concatStringsSep "" m else "313";
+      python = pkgs."python${versionStr}";
+      workspace = uv2nix.lib.workspace.loadWorkspace { workspaceRoot = root; };
+      overlay = workspace.mkPyprojectOverlay { sourcePreference = "wheel"; };
+      editableOverlay = workspace.mkEditablePyprojectOverlay { root = "$REPO_ROOT"; };
+
+      pythonSet =
+        (pkgs.callPackage pyproject-nix.build.packages {
+          inherit python;
+        }).overrideScope
+          (
+            lib.composeManyExtensions [
+              pyproject-build-systems.overlays.default
+              overlay
+            ]
+          );
+
+      # reuquired to create editable venv
+      editablePythonSet = pythonSet.overrideScope (
+        lib.composeManyExtensions [
+          editableOverlay
+          (final: prev: {
+            juspay-mcp = prev.juspay-mcp.overrideAttrs (old: {
+              src = lib.fileset.toSource {
+                root = old.src;
+                fileset = lib.fileset.unions [
+                  (old.src + "/pyproject.toml")
+                  (old.src + "/README.md")
+                  (old.src + "/juspay_mcp")
+                  (old.src + "/juspay_dashboard_mcp")
+                ];
+              };
+              nativeBuildInputs =
+                old.nativeBuildInputs
+                ++ final.resolveBuildSystem {
+                  editables = [ ];
+                };
+            });
+          })
+        ]
+      );
+      venv = editablePythonSet.mkVirtualEnv "juspay-mcp" workspace.deps.all;
+    in
+    {
+      packages = {
+        default = pythonSet.mkVirtualEnv "juspay-mcp" workspace.deps.all;
+        test = python.withPackages (ps: with ps; [ pytest ]);
+        stdio = pkgs.writeShellApplication {
+          name = "stdio";
+          runtimeInputs = [ venv ];
+          text = ''
+            exec "${venv}/bin/python" ${inputs.self}/juspay_mcp/stdio.py "$@"
+          '';
+        };
+      };
+      apps = {
+        default.program = "${self'.packages.default}/bin/juspay-mcp";
+        test.program = "${self'.packages.test}/bin/pytest";
+        stdio.program = "${self'.packages.stdio}/bin/stdio";
+      };
+
+
+      devShells = {
+        default = self'.devShells.uv2nix;
+
+        uv2nix = pkgs.mkShell {
+          inputsFrom = [
+            config.pre-commit.devShell
+          ];
+          packages = [
+            venv
+            pkgs.uv
+          ];
+          env = {
+            UV_NO_SYNC = "1";
+            UV_PYTHON = "${venv}/bin/python";
+            UV_PYTHON_DOWNLOADS = "never";
+          };
+
+          shellHook = # sh
+            ''
+              unset PYTHONPATH
+              export REPO_ROOT="$(git rev-parse --show-toplevel)"
```

Comment:
Why do we need this? What uses REPO_ROOT?

Reply:
https://github.com/juspay/python-nix-template/blob/f17c84c32359795a42ce5dc46364b0170a95028c/flake.nix#L210-L211